pilot/pkg/ambient/controller/aggregate.go

```diff
@@ -118,21 +119,26 @@ func (a *Aggregate) clusterAdded(cluster *multicluster.Cluster, stop <-chan stru
 
 	// don't modify remote clusters, just find their waypoint proxies and Pods
 	opts.LocalCluster = a.localCluster == cluster.ID
-
-	a.clusters[cluster.ID] = initForCluster(&opts)
+	a.clusters[cluster.ID] = initForCluster(opts)
 	return nil
 }
 
-func initForCluster(opts *Options) *ambientController {
+func initForCluster(opts Options) *ambientController {
 	if opts.LocalCluster {
-		// TODO handle istiodless remote clusters
-		initAutolabel(opts)
-		go func() {
-			if crdclient.WaitForCRD(gvk.KubernetesGateway, opts.Stop) {
-				waypointController := NewWaypointProxyController(opts.Client, opts.ClusterID, opts.WebhookConfig)
-				waypointController.Run(opts.Stop)
-			}
-		}()
+		election := leaderelection.
+			NewLeaderElection(opts.SystemNamespace, "pod name ", leaderelection.NamespaceController, "", opts.Client).
```

Comment:
`"pod name "` shouldn't be a constant but derived from `args.PodName`

Reply:
I think we want `revision` as well

Reply:
Yes, i haven't done that